exec/java-exec/src/main/java/org/apache/drill/exec/store/pcap/decoder/Packet.java

```
@@ -0,0 +1,371 @@
+        return getPort(136);
+      }
+      return getPort(40);
+    }
+    return getPort(0);
+  }
+
+  public int getDst_port() {
+    if (isPPPoV6Packet()) {
+      return getPort(66);
+    }
+    if (isIpV6Packet()) {
+      if (isRoutingV6) {
+        return getPort(138);
+      }
+      return getPort(42);
+    }
+    return getPort(2);
+  }
+
+  public byte[] getData() {
+    int payloadDataStart = getIPHeaderLength();
+    if (isTcpPacket()) {
+      payloadDataStart += this.getTCPHeaderLength(raw);
+    } else if (isUdpPacket()) {
+      payloadDataStart += this.getUDPHeaderLength();
+    } else {
+      return null;
+    }
+    byte[] data = null;
+    if (packetLength >= payloadDataStart) {
+      data = new byte[packetLength - payloadDataStart];
+      System.arraycopy(raw, ipOffset + payloadDataStart, data, 0, data.length);
+    }
+    return data;
+  }
+
+  private InetAddress getIPAddress(final boolean src) {
+    byte[] ipBuffer = getIpAddressBytes(src);
+    if (ipBuffer == null) {
+      return null;
+    }
+    try {
+      return InetAddress.getByAddress(ipBuffer);
+    } catch (UnknownHostException e) {
+      return null;
+    }
+  }
+
+  private byte[] getIpAddressBytes(final boolean src) {
+    int srcPos;
+    byte[] ipBuffer;
+    if (isIpV4Packet()) {
+      ipBuffer = new byte[4];
+      srcPos = src ? PacketConstants.IP4_SRC_OFFSET : PacketConstants.IP4_DST_OFFSET;
+    } else if (isIpV6Packet()) {
+      ipBuffer = new byte[16];
+      if (isRoutingV6) {
+        srcPos = src ? PacketConstants.IP6_SRC_OFFSET + 96 : PacketConstants.IP6_DST_OFFSET + 96;
+      } else {
+        srcPos = src ? PacketConstants.IP6_SRC_OFFSET : PacketConstants.IP6_DST_OFFSET;
+      }
+    } else if (isPPPoV6Packet()) {
+      ipBuffer = new byte[16];
+      srcPos = src ? PacketConstants.IP6_SRC_OFFSET + PacketConstants.PPPoV6_IP_OFFSET : PacketConstants.IP6_DST_OFFSET + PacketConstants.PPPoV6_IP_OFFSET;
+    } else {
+      return null;
```

What does a null value returned from here mean? An unsupported packet type? If so, caller should throw an UnsupportedXYZ exception, perhaps?